test/clojure/core_test/parse_boolean.cljc

```diff
@@ -0,0 +1,38 @@
+(ns clojure.core-test.parse-boolean
+  (:require clojure.core
+            [clojure.test :as t :refer [deftest testing is are]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer)  [when-var-exists]]))
+
+(when-var-exists clojure.core/parse-boolean
+  (deftest test-parse-boolean
+    (testing "common"
+      (are [expected x] (= expected (parse-boolean x))
+           false "false"
+           true  "true"
+           nil   ""
+           nil   "foo"
+           nil   "False"
+           nil   "FALSE"
+           nil   "True"
+           nil   "TRUE"))
```

Comment:
These parsing functions are quite involved, so I think we can have quite a bit more rigor here.

Would be good to have edge case tests for prefix/suffix (i.e. `ttrue` and `truee`). Also good to have tests for number strings (since `1` is commonly convertible to bool). Also, we should test for how whitespace is handled (on either end, and in the middle).